custom_components/wellbeing/vacuum.py

```diff
@@ -0,0 +1,158 @@
+"""Vacuum platform for Wellbeing."""
+
+import asyncio
+import logging
+import math
+
+from homeassistant.components.vacuum import StateVacuumEntity, VacuumEntityFeature
+from homeassistant.components.vacuum import (
+    STATE_CLEANING,
+    STATE_PAUSED,
+    STATE_RETURNING,
+    STATE_IDLE,
+    STATE_DOCKED,
+    STATE_ERROR,
+)
+from homeassistant.const import Platform
+from homeassistant.util.percentage import (
+    percentage_to_ranged_value,
+    ranged_value_to_percentage,
+)
+
+from . import WellbeingDataUpdateCoordinator
+from .const import DOMAIN
+from .entity import WellbeingEntity
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+SUPPORTED_FEATURES = (
+    VacuumEntityFeature.START
+    | VacuumEntityFeature.STOP
+    | VacuumEntityFeature.PAUSE
+    | VacuumEntityFeature.RETURN_HOME
+    | VacuumEntityFeature.BATTERY
+)
+
+VACUUM_STATES = {
+    1: STATE_CLEANING,  # Regular Cleaning
+    2: STATE_PAUSED,
+    3: STATE_CLEANING,  # Stop cleaning
+    4: STATE_PAUSED,  # Pause Spot cleaning
+    5: STATE_RETURNING,
+    6: STATE_PAUSED,  # Paused returning
+    7: STATE_RETURNING,  # Returning for pitstop
+    8: STATE_PAUSED,  # Paused returning for pitstop
+    9: STATE_DOCKED,  # Charging
+    10: STATE_IDLE,
+    11: STATE_ERROR,
+    12: STATE_DOCKED,  # Pitstop
+    13: STATE_IDLE,  # Manual stearing
+    14: STATE_IDLE,  # Firmware upgrading
+}
+
+BATTERY_LEVELS = {
+    0: 0,
+    1: 10,
+    2: 30,
+    3: 50,
+    4: 70,
+    5: 90,
+    6: 100,
+}
+
+BATTERY_ICONS = {
+    0: "mdi:battery-alert-variant-outline",
+    1: "mdi:battery-10",
+    2: "mdi:battery-30",
+    3: "mdi:battery-50",
+    4: "mdi:battery-70",
+    5: "mdi:battery-90",
+    6: "mdi:battery",
+}
+
+
+FAN_SPEEDS = {
+    1: "Quiet",
+    2: "Smart",
+    3: "Power",
+}
```

Comment:
Make a property of this in Appliance

Reply:
The specific set of values probably needs to be looked at before enabling the FAN_SPEED to ensure compatibility with both the original purei9 and the purei9.2.